src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -643,6 +632,18 @@ public Void caseTemplatizedType(TemplatizedType type) {
             emit("[]");
             return null;
           }
+          // Arguments<?> and NodeList<?> in es3 externs are correspondinly
+          // IArguments and NodeList interfaces (not-parametrized) in lib.d.ts.
+          // New* are temporary work-arounds for upstream externs.
+          // TODO(rado): upgrade closure compiler and remove them.
+          if (type.getDisplayName().equals("Arguments") || type.getDisplayName().equals("NewArguments")) {
```

Comment:
Unless I am mistaken `displayName` is qualified, so there is little chance that would overlap as custom classes will be `foo.bar.Arguments`. 

Also, I can't seem to find an api to get the originating file of the type declaration. I guess I can add some JSDoc heuristics, but then we are one reformating away from braking.